phpmyfaq/src/phpMyFAQ/Administration/Backup.php

```diff
@@ -153,4 +157,38 @@ private function getBackupHeader(string $tableNames): array
             '-- Otherwise this backup will be broken.'
         ];
     }
+
+    /**
+     * Creates a ZipArchive of the content-folder
+     *
+     * @throws \Exception
+     */
+    public function createContentFolderBackup(): string|bool
+    {
+        $zipFile = PMF_ROOT_DIR . DIRECTORY_SEPARATOR . 'content.zip';
+
+        $zipArchive = new ZipArchive();
+        if ($zipArchive->open($zipFile, ZipArchive::CREATE) !== true) {
+            throw new Exception('Error while creating ZipArchive');
+        }
+
+        $files = new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator(PMF_CONTENT_DIR)
+        );
+
+        foreach ($files as $file) {
+            $filePath = $file->getRealPath();
+            $relativePath = substr($filePath, strlen(PMF_CONTENT_DIR) + 1);
+
+            if ($file->isDir()) {
+                $zipArchive->addEmptyDir($relativePath);
+            } else {
+                $zipArchive->addFile($filePath, $relativePath);
+            }
+        }
+
+        $zipArchive->close();
+
+        return file_exists($zipFile) ? $zipFile : false;
```

Comment:
Here's another one which should be an exception instead of a bool value.

Reply:
Fully not necessary, at this time. I changed it.